docs/playbook/unit-testing-guide.md

```diff
@@ -33,25 +33,35 @@ the behaviours of that [with tests][tests_example].
-Start by creating your own glue job script as usual, `script_name.py`, and also a test file `test_script_name.py`.  
-Test files should be named with "test" at the beginning of the file. For example, `test_cleaning_script.py`.
-  - You will need to organise your script according to the [Organising your code so it can be tested](#organising-your-code-so-it-can-be-tested) above
-- Define a function within `script_name.py`, which takes in the input DataFrame(s), and returns the DataFrame to be written.
-- Within your `test_script_name.py` start creating tests. 
-  You can use the [test_spark_example][test_spark_example] as a template and [address cleaning][address_cleaning] as an example of a fully tested script. 
+- There are 2 scenarios: 
+  - In the ideal scenario, you should be writing your script and your tests at the same time. Start by creating your own glue job script as usual, `script_name.py`, and also a test file `test_script_name.py`. Test files should be named with "test" at the beginning of the file. For example, `test_cleaning_script.py`.
+  - In the non-ideal scenario, you are writing tests at posteriori for a script that already exists. Pick this script and create the corresponding test file named `test_script_name.py`.
+
+- You will need to organise your script (or reorganise it if it already exists) according to the [Organising your code so it can be tested](#organising-your-code-so-it-can-be-tested) above. For existing scripts, it includes adding the line
+` __name__ = “__main__”:`
+Just after the methods definitions and just before the part that reads arguments from the AWS environment (look at examples). Indent everything under this new line.
+- Define a function within `script_name.py`, which takes in the input DataFrame(s), and returns the DataFrame to be written. (For existing scripts: take the processing code out of the main part and turn it into one or several methods).
+- Within your `test_script_name.py` start creating tests. You can use the [test_spark_example][test_spark_example] as a template and [address cleaning][address_cleaning] as an example of a fully tested script. If you do, rename the class name into something relevant, for instance TestMechFireDpaCleaning. 
   Test method names should start with a "test" in their name, otherwise the tests won’t be run. 
   Ensure the name clearly describes what is being tested. 
   For example, if you are testing the behaviour of adding a unique id column, then a suitable name may be along the lines of `test_creates_unique_id_column`.
-- There are some helper functions in `unit_testing_helpers.py` which you can use in your tests. 
+- There are some helper functions in `unit_testing_helpers.py` which you can use in your tests. For example, `assertDictionaryContains` is a useful function that asserts that a specific field and value exist in an output, instead of asserting equality for a full data line.
```

Comment:
```suggestion
- There are some helper functions in `unit_testing_helpers.py` which you can use in your tests. For example, `assertDictionaryContains` is a useful function that asserts that specific fields and values exist in an output, instead of asserting equality for a full data line.
```